Apply all Buildroot changes until Feb 23th 2010.

Only needed for Buildroot < 2010.02

Index: buildroot/package/imagemagick/imagemagick.mk
===================================================================
--- buildroot.orig/package/imagemagick/imagemagick.mk	2010-03-18 21:01:41.000000000 +0100
+++ buildroot/package/imagemagick/imagemagick.mk	2010-03-19 09:09:52.000000000 +0100
@@ -3,14 +3,13 @@
 # imagemagick
 #
 #############################################################
-IMAGEMAGICK_VERSION:=6.3.7
-IMAGEMAGICK_SOURCE:=ImageMagick.tar.bz2
+IMAGEMAGICK_MAJOR:=6.4.8
+IMAGEMAGICK_VERSION:=$(IMAGEMAGICK_MAJOR)-4
+IMAGEMAGICK_SOURCE:=ImageMagick-$(IMAGEMAGICK_VERSION).tar.bz2
 IMAGEMAGICK_SITE:=ftp://ftp.imagemagick.org/pub/ImageMagick
 IMAGEMAGICK_DIR:=$(BUILD_DIR)/ImageMagick-$(IMAGEMAGICK_VERSION)
 IMAGEMAGICK_CAT:=$(BZCAT)
-#IMAGEMAGICK_BINARY:=convert
-#IMAGEMAGICK_TARGET_BINARY:=usr/bin/$(IMAGEMAGICK_BINARY)
-IMAGEMAGICK_LIB:=$(TARGET_DIR)/usr/lib/libMagick.so
+IMAGEMAGICK_LIB:=$(TARGET_DIR)/usr/lib/libMagickCore.so
 
 IMAGEMAGICK_TARGET_BINARIES:=$(TARGET_DIR)/usr/bin/animate
 IMAGEMAGICK_TARGET_BINARIES+=$(TARGET_DIR)/usr/bin/compare
@@ -24,19 +23,25 @@
 
 IMAGEMAGICK_COPY:=cp -df --preserve=mode,ownership
 $(DL_DIR)/$(IMAGEMAGICK_SOURCE):
-	$(WGET) -P $(DL_DIR) $(IMAGEMAGICK_SITE)/$(IMAGEMAGICK_SOURCE)
+	$(call DOWNLOAD,$(IMAGEMAGICK_SITE),$(IMAGEMAGICK_SOURCE))
 
 $(IMAGEMAGICK_DIR)/.unpacked: $(DL_DIR)/$(IMAGEMAGICK_SOURCE)
 	$(IMAGEMAGICK_CAT) $(DL_DIR)/$(IMAGEMAGICK_SOURCE) | tar -C $(BUILD_DIR) $(TAR_OPTIONS) -
-	toolchain/patch-kernel.sh $(IMAGEMAGICK_DIR) package/imagemagick/ imagemagick-\*.patch\*
+	toolchain/patch-kernel.sh $(IMAGEMAGICK_DIR) package/imagemagick/ imagemagick-$(IMAGEMAGICK_VERSION)\*.patch\*
 	$(CONFIG_UPDATE) $(IMAGEMAGICK_DIR)/config
 	touch $@
 
+ifeq ($(BR2_LARGEFILE),y)
+IMAGEMAGICK_CONF_OPTS = ac_cv_sys_file_offset_bits=64
+else
+IMAGEMAGICK_CONF_OPTS = ac_cv_sys_file_offset_bits=32
+endif
+
 $(IMAGEMAGICK_DIR)/.configured: $(IMAGEMAGICK_DIR)/.unpacked
 	(cd $(IMAGEMAGICK_DIR); rm -f config.cache; \
 		$(TARGET_CONFIGURE_OPTS) \
 		$(TARGET_CONFIGURE_ARGS) \
-		./configure \
+		./configure $(QUIET) \
 		--target=$(GNU_TARGET_NAME) \
 		--host=$(GNU_TARGET_NAME) \
 		--build=$(GNU_HOST_NAME) \
@@ -56,6 +61,7 @@
 		--without-fpx \
 		--without-freetype \
 		--without-x \
+		$(IMAGEMAGICK_CONF_OPTS) \
 	)
 	touch $@
 
@@ -63,22 +69,22 @@
 	$(MAKE) -C $(IMAGEMAGICK_DIR)
 	touch $@
 
-$(STAGING_DIR)/usr/lib/libMagick.a: $(IMAGEMAGICK_DIR)/.compiled
+$(STAGING_DIR)/usr/lib/libMagickCore.a: $(IMAGEMAGICK_DIR)/.compiled
 	$(MAKE) DESTDIR=$(STAGING_DIR) -C $(IMAGEMAGICK_DIR) install
 	touch -c $@
 
-$(IMAGEMAGICK_LIB): $(STAGING_DIR)/usr/lib/libMagick.a
-	$(IMAGEMAGICK_COPY) $(STAGING_DIR)/usr/lib/libWand.so* $(TARGET_DIR)/usr/lib/
-	-$(STRIPCMD) $(STRIP_STRIP_UNNEEDED) $(TARGET_DIR)/usr/lib/libWand.so*
-	mkdir -p $(TARGET_DIR)/usr/lib/ImageMagick-$(IMAGEMAGICK_VERSION)
-	$(IMAGEMAGICK_COPY) -r $(STAGING_DIR)/usr/lib/ImageMagick-$(IMAGEMAGICK_VERSION) $(TARGET_DIR)/usr/lib
-	$(IMAGEMAGICK_COPY) $(STAGING_DIR)/usr/lib/libMagick.so* $(TARGET_DIR)/usr/lib/
+$(IMAGEMAGICK_LIB): $(STAGING_DIR)/usr/lib/libMagickCore.a
+	$(IMAGEMAGICK_COPY) $(STAGING_DIR)/usr/lib/libMagickWand.so* $(TARGET_DIR)/usr/lib/
+	-$(STRIPCMD) $(STRIP_STRIP_UNNEEDED) $(TARGET_DIR)/usr/lib/libMagickWand.so*
+	mkdir -p $(TARGET_DIR)/usr/lib/ImageMagick-$(IMAGEMAGICK_MAJOR)
+	$(IMAGEMAGICK_COPY) -r $(STAGING_DIR)/usr/lib/ImageMagick-$(IMAGEMAGICK_MAJOR) $(TARGET_DIR)/usr/lib
+	$(IMAGEMAGICK_COPY) $(STAGING_DIR)/usr/lib/libMagickCore.so* $(TARGET_DIR)/usr/lib/
 	-$(STRIPCMD) $(STRIP_STRIP_UNNEEDED) $(IMAGEMAGICK_LIB)*
 	touch -c $@
 
 $(IMAGEMAGICK_DIR)/.libinstall: $(IMAGEMAGICK_LIB)
-	libtool --finish $(TARGET_DIR)/usr/lib/ImageMagick-6.3.5/modules-Q16/coders
-	libtool --finish $(TARGET_DIR)/usr/lib/ImageMagick-6.3.5/modules-Q16/filters
+	$(IMAGEMAGICK_DIR)/libtool --finish $(TARGET_DIR)/usr/lib/ImageMagick-$(IMAGEMAGICK_MAJOR)/modules-Q16/coders
+	$(IMAGEMAGICK_DIR)/libtool --finish $(TARGET_DIR)/usr/lib/ImageMagick-$(IMAGEMAGICK_MAJOR)/modules-Q16/filters
 	touch $@
 
 $(TARGET_DIR)/usr/bin/animate: $(IMAGEMAGICK_LIB)
@@ -127,25 +133,20 @@
 	-$(STRIPCMD) $(STRIP_STRIP_UNNEEDED) $(TARGET_DIR)/usr/bin/convert
 	touch $@
 
-imagemagick: uclibc jpeg tiff $(IMAGEMAGICK_LIB) \
+imagemagick: jpeg tiff $(IMAGEMAGICK_LIB) \
 		$(IMAGEMAGICK_DIR)/.libinstall \
 		$(IMAGEMAGICK_TARGET_BINARIES)
 
 imagemagick-source: $(DL_DIR)/$(IMAGEMAGICK_SOURCE)
 
+imagemagick-unpacked:$(IMAGEMAGICK_DIR)/.unpacked
+
 imagemagick-clean:
-	rm -f $(TARGET_DIR)/$(IMAGEMAGICK_TARGET_BINARY)
-	rm -f $(TARGET_DIR)/usr/bin/animate
-	rm -f $(TARGET_DIR)/usr/bin/compare
-	rm -f $(TARGET_DIR)/usr/bin/composite
-	rm -f $(TARGET_DIR)/usr/bin/conjure
-	rm -f $(TARGET_DIR)/usr/bin/convert
-	rm -f $(TARGET_DIR)/usr/bin/display
-	rm -f $(TARGET_DIR)/usr/bin/import
-	rm -f $(TARGET_DIR)/usr/bin/mogrify
-	rm -f $(TARGET_DIR)/usr/bin/montage
-	rm -rf $(TARGET_DIR)/usr/lib/ImageMagick-$(IMAGEMAGICK_VERSION)
-	rm -rf $(TARGET_DIR)/usr/lib/ImageMagick-$(IMAGEMAGICK_VERSION)
+	for target_binary in $(IMAGEMAGICK_TARGET_BINARIES); do \
+		rm -f $$target_binary; \
+	done
+	rm -rf $(TARGET_DIR)/usr/lib/libMagick*
+	rm -rf $(TARGET_DIR)/usr/lib/ImageMagick-$(IMAGEMAGICK_MAJOR)
 	-$(MAKE) -C $(IMAGEMAGICK_DIR) clean
 
 imagemagick-dirclean:
@@ -155,6 +156,6 @@
 # Toplevel Makefile options
 #
 #############################################################
-ifeq ($(strip $(BR2_PACKAGE_IMAGEMAGICK)),y)
+ifeq ($(BR2_PACKAGE_IMAGEMAGICK),y)
 TARGETS+=imagemagick
 endif
